x-pack/plugins/cases/public/components/user_action_tree/user_action_markdown.test.tsx

```diff
@@ -37,16 +38,38 @@ describe('UserActionMarkdown ', () => {
     expect(wrapper.find(`[data-test-subj="markdown-link"]`).first().text()).toContain('hyperlink');
   });
 
-  it('Save button click calls onSaveContent and onChangeEditable', async () => {
+  it('Save button click calls onSaveContent and onChangeEditable when text area value changed', async () => {
     const wrapper = mount(
       <TestProviders>
         <UserActionMarkdown {...defaultProps} />
       </TestProviders>
     );
+
+    wrapper
+      .find(`.euiMarkdownEditorTextArea`)
+      .first()
+      .simulate('change', {
+        target: { value: newValue },
+      });
+
     wrapper.find(`[data-test-subj="user-action-save-markdown"]`).first().simulate('click');
 
     await waitFor(() => {
-      expect(onSaveContent).toHaveBeenCalledWith(defaultProps.content);
+      expect(onSaveContent).toHaveBeenCalledWith(newValue);
+      expect(onChangeEditable).toHaveBeenCalledWith(defaultProps.id);
+    });
+  });
+  it('Does not call onSaveContent if no change from current text', async () => {
+    const wrapper = mount(
+      <TestProviders>
+        <UserActionMarkdown {...defaultProps} />
+      </TestProviders>
+    );
+
+    wrapper.find(`[data-test-subj="user-action-save-markdown"]`).first().simulate('click');
+
+    await waitFor(() => {
+      expect(onSaveContent).not.toHaveBeenCalled();
```

Comment:
NIT: following the implementation, maybe it would be a little bit better to test with _waitFor_ `onChangeEditable.toHaveBeenCalled` only, and then, later, check `onSaveContent.not.toHaveBeenCalled` outside the _waitFor_